File: CellsToServers/CellsToServersApp/ArrayPartition/Coords.cs
using System;
using System.Text;

namespace CellsToServersApp.ArrayPartition
{
    public class Coords
    {
        public int[] IndicesArray { get; set; }
        public int HeftOfRegion { get; set; }
        public int MaxCellValue { get; set; }
        public int[] MaxCellIndices { get; set; }
        public int[] CenterOfMassIndices { get; set; }
        public int CrossBorderHeft { get; set; }

        public double differenceFromDelta(double delta)
        {
            return delta - HeftOfRegion;
        }

        public void printCoords(int spaceDimension, int serialNO)
        {
            Console.Write("{0}. tile: [", serialNO);
            for (int idx = 0; idx < spaceDimension; idx++)
            {
                Console.Write(" {0} {1} ", IndicesArray[2 * idx], IndicesArray[2 * idx + 1]);
            }
            Console.Write("] : {0} heft; max cell value: {1}, max cell indices:", HeftOfRegion, MaxCellValue);
            foreach (var idx in MaxCellIndices)
            {
                Console.Write(" " + idx);
            }
            Console.Write(", center of mass indices:");
            foreach (var idx in CenterOfMassIndices)
            {
                Console.Write(" " + idx);
            }
            Console.WriteLine(", cross border heft: {0}", CrossBorderHeft);
        }

        public void writeToStringBuilder(int spaceDimension, StringBuilder strBldr)
        {
            strBldr.Append(HeftOfRegion);
            for (int idx = 0; idx < spaceDimension; idx++)
            {
                strBldr.Append(" ").Append(IndicesArray[2 * idx]).Append(" ").Append(IndicesArray[2 * idx + 1]);
            }
            strBldr.AppendLine();
        }
    }
}

File: CellsToServersCppLinux/HierarchicalTilingAppCpp/ArrayPartition/Coords.cpp
#include "Coords.hpp"

namespace ArrayPartition
{
	int * Coords::getExtendedIndicesArray() const
	{
		return extendedIndicesArray;
	}

	void Coords::setExtendedIndicesArray(int * inputExtendedIndicesArray)
	{
		this->extendedIndicesArray = inputExtendedIndicesArray;
	}

	int Coords::getHeftOfRegion() const
	{
		return heftOfRegion;
	}

	void Coords::setHeftOfRegion(int inputHeftOfRegion)
	{
		this->heftOfRegion = inputHeftOfRegion;
	}

	double Coords::differenceFromDelta(double delta)
	{
		return abs(delta - heftOfRegion);
	}

	void Coords::printCoords(int spaceDimension, int serialNO)
	{
		std::cout << serialNO << ". tile: [";
		for (int idx = spaceDimension - 1; idx >= 0; idx--)
		{
			std::cout << " " << extendedIndicesArray[2 * idx + 1] << " " 
				<< extendedIndicesArray[2 * idx + 2] << " ";
		}
		std::cout << "] : " << heftOfRegion << " heft" << std::endl;
	}

	void Coords::writeToStringBuilder(int spaceDimension, std::stringstream& strBldr)
	{
		strBldr << heftOfRegion;
		for (int idx = spaceDimension - 1; idx >= 0; idx--)
		{
			strBldr << " " << extendedIndicesArray[2 * idx + 1] << " "
				<< extendedIndicesArray[2 * idx + 2];
		}
		strBldr << std::endl;
	}
}